xarray/core/dataset.py

```diff
@@ -333,7 +338,13 @@ class Dataset(Mapping, ImplementsDatasetReduce, DataWithCoords):
     _coarsen_cls = rolling.DatasetCoarsen
     _resample_cls = resample.DatasetResample
 
-    def __init__(self, data_vars=None, coords=None, attrs=None,
+    def __init__(self,
+                 data_vars: Optional[Mapping[Hashable, Union[
+                     'DataArray', Variable,
+                     Tuple[Hashable, Any],
+                     Tuple[Tuple[Hashable, ...], Any]]]] = None,
```

Comment:
This one also can handle any tuples `args` that you can use like `Variable(*args)` (maybe we could define a `VariableArgs` alias of some sort?)

Reply:
Yes I added a comment re VariableArgs, def would be good to move some of these long definitions into a single place (but if you're OK with leaving until the next PR let's do that)